src/map/index.ts

```diff
@@ -11,24 +14,58 @@ export interface Country {
 }
 
 let countries: Map<number, Country> | undefined = undefined;
+let countryNameLut: Map<string, number> | undefined = undefined;
+
+let preparedCountries: Prepared[] | undefined = undefined;
 
 export const getCountries = async () => {
     if (!countries) {
         const world = await getWorld(config.apiKey);
 
         countries = new Map(
             world.map(({ name, id, coordinates }) => [
-                id,
+                parseInt(id),
                 {
                     name,
-                    id,
+                    id: parseInt(id),
                     coordinates,
                 },
             ])
         );
+
+        countryNameLut = new Map(
+            world.map(({ name, id }) => [name, parseInt(id)])
+        );
     }
 
     return countries;
 };
 
+export interface SearchMatch {
+    value: number;
+    name: string;
+}
+
+export const fuzzySearchCountries = async (
+    input: string
+): Promise<SearchMatch[]> => {
+    if (!preparedCountries) {
+        preparedCountries = await getCountries().then((cs) =>
+            [...cs.values()].map((c) => prepare(c.name))
+        );
+    }
+
+    /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+    const results = go(input, preparedCountries!, {
```

Comment:
theres not a way to do this without making eslint mad? seems like there should be since its not a weird hack like the decorators thing

Reply:
It is bad style. I overrode it here because there's no way this can go wrong in this case and working around it would be annoying, but it is a bit iffy

Reply:
seems to me like if you are gonna enforce linting its silly to just break the rule because its annoying lol, but ok

Reply:
To circumvent this properly, I would have to throw an Error for a scenario that can never actually happen, so I decided to just say fuck it in this case